src/screens/DiscoveryScreenV2/hooks/useDiscovery.js

```diff
@@ -21,17 +26,43 @@ const useDiscovery = () => {
       });
 
       await getFollowingTopic().then((response) => {
-        console.log({response}, 'likak');
         following.setFollowingTopics(response.data, followingDispatch);
       });
       setRefreshing(false);
     } catch (e) {
       setRefreshing(false);
     }
   };
+
+  const onRefreshDiscoveryUser = async () => {
+    try {
+      const discoveryInitialUserResponse = await DiscoveryRepo.fetchInitialDiscoveryUsers();
+      DiscoveryAction.setDiscoveryInitialUsers(
+        discoveryInitialUserResponse.suggestedUsers,
+        discoveryDispatch
+      );
+    } catch (e) {
+      setRefreshing(false);
+    }
+  };
+
+  const handleUpdateDiscoveryUser = (id, isFollow) => {
+    if (users && Array.isArray(users)) {
+      const mapUser = users.map((user) => {
+        if (user.user_id === id) {
+          return {...user, user_id_follower: isFollow ? profile?.myProfile?.user_id : null};
+        }
+        return {...user};
+      });
+      DiscoveryAction.setDiscoveryInitialUsers(mapUser, discoveryDispatch);
+    }
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In the `handleUpdateDiscoveryUser` function, you are creating a new array of users every time this function is called, even if the user id does not match any of the existing users. This could lead to unnecessary memory usage and performance degradation when dealing with large arrays. Consider checking if the user exists before mapping over the entire array.
```diff
    if (users && Array.isArray(users)) {
      const userExists = users.some(user => user.user_id === id);
      if (userExists) {
        const mapUser = users.map((user) => {
          if (user.user_id === id) {
            return {...user, user_id_follower: isFollow ? profile?.myProfile?.user_id : null};
          }
          return {...user};
        });
        DiscoveryAction.setDiscoveryInitialUsers(mapUser, discoveryDispatch);
      }
    }
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->